skbio/sequence/_grammared_sequence.py

```diff
@@ -209,6 +211,21 @@ def _noncanonical_codes(cls):
             cls.__noncanonical_codes = np.asarray([ord(c) for c in noncanonical_chars])
         return cls.__noncanonical_codes
 
+    @classproperty
+    def _degenerate_hash(cls):
+        if cls.__degenerate_hash is None:
+            htab = np.zeros((256,), dtype=np.uint8)
+            htab[cls._degenerate_codes] = 1
+            cls.__degenerate_hash = htab
+        return cls.__degenerate_hash
+
+    @classproperty
+    def _degen_nonca_hash(cls):
```

Comment:
This place is very tricky: When you do `arr2 = arr1`, you are not copying all elements of `arr1`, but simply creating a reference to `arr1`. As a consequence, when you modify the content of `arr2`, you will find that `arr1` is also modified. This can be a problem. The solution is to do `arr2 = arr1.copy()`. [Here](https://www.educative.io/answers/pass-by-value-vs-pass-by-reference) is some reading materials on this.

Reply:
Thank you!